src/form-generator-one-step.ts

```diff
@@ -37,15 +37,22 @@ export class FormGeneratorImpl implements FormGenerator {
         this.formElementCreator = new FormElementCreator(this.elementCreator);
     }
 
-    generateForm(): HTMLFormElement {
-        const form = this.elementCreator.createElement('form') as HTMLFormElement;
+    generateForm = async (): Promise<HTMLFormElement> => {
+        const form = document.createElement('form');
+        document.body.appendChild(form);
+
         const elements = [...this.config.selects, ...this.config.fields, ...this.config.data, ...this.config.buttons];
 
-        elements.forEach((element) => {
-            this.formElementCreator.createFormElement(form, element);
-        });
+        for (const element of elements) {
+            if (element.name === 'Phone') {
+                await this.formElementCreator.createPhoneInput(form, element);
+            } else {
+                this.formElementCreator.createFormElement(form, element);
+            }
+        }
+
```

Comment:
mam wrazenei ze ten kawalek kodu juz gdzies sie powtarzal a skoro tak jest to moze warto zrobic z tego funkcje w jakiejs klasie typu form-utils.ts 
No i warto zaaplikowac moje uwagi z poprzedniego miejsca do tego rowniez. Niezaleznie czy strategia sie do tego nada czy nie